plugin-hrm-form/src/utils/sharedState.js

```
@@ -11,7 +15,7 @@ export const saveFormSharedState = async (form, task) => {
   if (!featureFlags.enable_transfers) return null;
 
   try {
-    if (sharedStateClient === undefined || sharedStateClient.connectionState !== 'connected') {
+    if (!isSharedStateClientConnected(sharedStateClient)) {
       console.error('Error with Sync Client conection. Sync Client object is: ', sharedStateClient);
       window.alert(strings.SharedStateSaveFormError);
```

Should we be firing alerts & swallowing exceptions this low down the stack? Would it not be better to let the exception be thrown and let the caller  decide how to handle it?

---

I agree, throwing an exception and handling alerts at a higher step of the hierarchy would be a  better way to do this.

---

I agree with this, but what would be a better approach? The Sync Client can be initially OK but fail in further calls. Should we check for it's state prior to try any call that involves it? It's been a while since I haven't seen this code but I remember doing this as a way to at least know if something went wrong without "blocking the counselor workflow".

---

In case it goes inside this error clause, the function currently:
- logs the error
- calls window.alert
- returns null

It could be changed to:
- logs the error (which I think it's the correct place to do it)
- but not call window.alert
- and throw an error (e.g.: SyncClientConnectionError) instead of return null

At setupActions file, where we use this function, I think it would be a better place to window.alert.

current setupActions code:
```
...
const documentName = await saveFormSharedState(form, payload.task);
...
```
Could be changed to:
```
...
try {
  const documentName = await saveFormSharedState(form, payload.task);
  ...
} catch (err) {
  if (err instanceof SyncClientConnectionError) {
    window.alert();
  }
}
// maybe a finally clause if we need
```

Something like this I think would be ideal. But since this is not the most straightforward change to test, I'll focus on getting the functionality ready and thid PR merged, before refactoring the way we handle these error alerts.